src/core/meshcop/dataset_manager.cpp

```diff
@@ -492,9 +492,32 @@ ThreadError DatasetManager::Set(Coap::Header &aHeader, Message &aMessage, const
             aMessage.Read(offset, sizeof(Tlv), &data.tlv);
             aMessage.Read(offset + sizeof(Tlv), data.tlv.GetLength(), data.value);
 
-            if (data.tlv.GetType() != Tlv::kCommissionerSessionId)
+            switch (data.tlv.GetType())
             {
+            case Tlv::kCommissionerSessionId:
+                // do not store Commissioner Session ID TLV
+                break;
+
+            case Tlv::kDelayTimer:
+            {
+                DelayTimerTlv *delayTimerTlv = reinterpret_cast<DelayTimerTlv *>(&tlv);
+
+                if (doesAffectMasterKey && delayTimerTlv->GetDelayTimer() < DelayTimerTlv::kDelayTimerDefault)
+                {
+                    delayTimerTlv->SetDelayTimer(DelayTimerTlv::kDelayTimerDefault);
+                }
+                else if (doesAffectConnectivity && delayTimerTlv->GetDelayTimer() < DelayTimerTlv::kDelayTimerMinimal)
```

Comment:
Here, do we need to add `doesAffectConnectivity` check? According to spec, when pending dataset does not update the master key and carried delay timer value is less than `DELAY_TIMER_MINIMAL`, set the delay timer with `DELAY_TIMER_MINIMAL`. For example, pending dataset set request would change TLV that would not affect network connectivity.